src/transaction/handler/mod.rs

```diff
@@ -0,0 +1,44 @@
+use std::any::Any;
+
+use crate::{tx_builder::TxBuilderError, NetworkInfo, ScriptGroup};
+
+use self::sighash::Secp256k1Blake160SighashAllScriptContext;
+
+use super::builder::tx_data::TxData;
+
+pub mod sighash;
+
+pub trait ScriptHandler {
+    fn build_transaction(
```

Comment:
please add doc for this trait, for example, what's the diff of returning true or false, sdk user may use this trait to implement the customized script handler.